File: 01.fractal-julia/fractal_simd.cpp
#include "fractal_simd.h"

#include <complex>
#include <cstring>

extern int max_iteraciones;
extern std::complex<double> c;

void julia_simd(double x_min, double y_min, double x_max, double y_max, uint32_t width, uint32_t height, uint32_t *pixel_buffer)
{
    std::memset(pixel_buffer, 0xFF000000, width * height * sizeof(uint32_t));
}

File: 02.fractal-java/src/main/java/com/programacion/paralela/FractalSimd.java
package com.programacion.paralela;

import java.nio.ByteBuffer;

public class FractalSimd {
    public ByteBuffer pixelBuffer;

    public FractalSimd() {
        pixelBuffer = ByteBuffer.allocateDirect(FractalParams.WIDTH * FractalParams.HEIGHT * 4);
    }

    public void juliaSimd() {
        FractalDll.INSTANCE.julia_simd(
                FractalParams.xMin,
                FractalParams.yMin,
                FractalParams.xMax,
                FractalParams.yMax,
                FractalParams.WIDTH,
                FractalParams.HEIGHT,
                FractalParams.max_iteraciones,
                pixelBuffer
        );
    }
}
